test/metabase/api/common_test.clj

```diff
@@ -85,3 +89,49 @@
 
   (testing "multi values a vector as well"
     (is (= [1 2 3] (api/parse-multi-values-param ["1" "2" "3"] parse-long)))))
+
+;; set up for testing permission failure event publishing
+(def ^:dynamic *events* nil)
+
+(methodical/defmethod events/publish-event! ::permission-failure-event
+  [topic event]
+  (swap! *events* conj [topic event]))
+
+(deftest *-check-functions-publish-events
```

Comment:
Haven't seen the `*-` prefix before, is that to indicate the test is using a dynamic var instead of the real audit tables?

Reply:
I was using it as a placeholder, like "[something]-check" but I can see that it's confusing in this context - I'll change the name to just `check-functions-publish-events` instead.

Reply:
Ohh gotcha